src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -60,15 +64,35 @@ object RockSlicer {
       !block.faces.exists(_.processorJoint)
     }
 
-    // Process processor blocks before continueing with remaining blocks
-    // Collect all blocks that contain processor joints from all nodes
-    val allProcessorBlocks = processorBlocks.collect()
+    // Process processor blocks before continuing with remaining blocks
     // Search blocks for matching processor joints
-    if (allProcessorBlocks.length > 0) {
-      val reconstructedBlocks = mergeBlocks(allProcessorBlocks, Seq.empty[Block], globalOrigin)
+    if (processorBlocks.count() > 0) {
+      // Create pair RDD that contains initially all the blocks on each partion as the keys and
+      // an empty sequence of blocks as the values. The keys represent blocks that are yet to be merged,
```

Comment:
Nice, the comments make this a lot easier to understand. Good work.

However, is it really appropriate to use the terms "key" and "value" here?  It seems to me that we have a collection of pairs, but the first element in each pair isn't really a "key" that serves as an identifier for the "values" in the traditional case. I noticed these terms in spark discussions for pair RDDs, but my understanding is they're thinking about this more traditional situation that doesn't seem to apply to us.